plugins/inputs/openweathermap/openweathermap.go

```diff
@@ -91,26 +102,16 @@ func (n *OpenWeatherMap) Gather(acc telegraf.Accumulator) error {
 		n.client = client
 	}
 
-	units := n.Units
 	switch n.Units {
-	case "imperial", "standard":
-		break
+	case "imperial", "standard", "metric":
 	default:
-		units = defaultUnits
+		return fmt.Errorf("unknown units: %s", n.Units)
```

Comment:
No change required, but this may break someone's dependency on this feature, in this case we'll prefer less automagic.

Reply:
Yes.  If you had a typo in the units setting in your conf it would fall back to metric without telling the user.